plugin/audit_log/buffer.c

```diff
@@ -163,11 +167,25 @@ void audit_log_buffer_shutdown(audit_log_buffer_t *log)
 int audit_log_buffer_write(audit_log_buffer_t *log, const char *buf, size_t len)
 {
   if (len > log->size)
-    return(1);
+  {
+    if (!log->drop_if_full)
+    {
+      mysql_mutex_lock(&log->mutex);
```

Comment:
We can write message to the error log. Something like "audit_log_buffer size N is too small. Please consider increasing it to at least M.". We can write it every time when M breaks new record or once in a period or both.